server/utils/passport/local-signup.js

```diff
@@ -1,91 +1,63 @@
 import passport from 'passport'
+import moment from 'moment'
 import config from '../../configs/config'
 import { collections } from '../mongoCollections'
 import { hash } from '../crypto/hash'
-import defaultUserConfig from '../../configs/defaultUserConfig'
 import basePathConfig from '../../configs/basePathConfig'
+import {
+  DEFAULT_CONFIG_AT_SIGNUP,
+  SIGN_UP_UPDATE_DATA,
+  INITIAL_ROLE,
+} from '../../constants'
 
 const basePath = basePathConfig || ''
 
 export default (req, res, next) => {
   passport.authenticate(
     'local-signup',
     { session: true },
-    function (err, user, reqBody) {
-      if (err) {
-        return res.redirect(`${basePath}/login?e=${err}`)
-      }
-      if (user) {
-        return res.redirect(`${basePath}/signup?e=existingUser`)
-      }
-      const password = reqBody.password
-      const username = reqBody.username
-      const email = reqBody.email
-      const display_name = reqBody.display_name
-      const { appDb } = req.app.locals
-      const hashedPW = hash(password)
-      appDb.collection(collections.users).findOneAndUpdate(
-        { uid: username },
-        {
-          $set: {
-            display_name: display_name,
-            title: '',
-            department: '',
-            company: '',
-            mail: email,
-            member_of: '',
-            bad_pwd_count: 0,
-            lockout_time: 0,
-            last_logoff: Date.now(),
-            last_logon: Date.now(),
-            account_expires: null,
+    async function (err, user, reqBody) {
+      try {
+        if (err) return res.redirect(`${basePath}/login?e=${err}`)
+
+        if (user) return res.redirect(`${basePath}/signup?e=existingUser`)
+
+        const password = reqBody.password
+        const uid = reqBody.username
+        const email = reqBody.email
+        const display_name = reqBody.display_name
+        const { appDb } = req.app.locals
+        const hashedPW = hash(password)
+        const account_expires = moment().add(2, 'years').format()
```

Comment:
is 2 years a requirement for account expiry?

Reply:
There's a UI component where an Admin sets the account expiration. Working on that part now